plotter/heat.go

```diff
@@ -0,0 +1,210 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"image/color"
+	"math"
+
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/palette"
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+)
+
+// GridXYZ describes three dimensional data where the X and Y
+// coordinates are arranged on a rectangular grid.
+type GridXYZ interface {
+	// Dims returns the dimensions of the grid.
+	Dims() (c, r int)
```

Comment:
btw. It seems that you didn't get the last version I made: https://gist.github.com/egonelbre/5770469d6d7b955a4559 So you might be able to grab something from there as well. Although I haven't tested it performance wise.

And looking at the loop excision - it seems that it should remove the previous "seen" entries otherwise there might be a problem with:

    """v"""\
       /"v"V
       V\___

(Sorry for the curde graphics)... I.e. when you take out the first loop from the contour line, the next loop would connect to that loop. And, for me, it's not obvious how it will work in that case.